src/analysis/retail/deathknight/shared/ExterminateCostNormalizer.ts

```diff
@@ -0,0 +1,73 @@
+import SPELLS from 'common/SPELLS';
+import talents from 'common/TALENTS/deathknight';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { AnyEvent, CastEvent, HasAbility } from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+
+/**
+ * The "Exterminate" buff reduces Obliterate/Marrowrend rune costs by 100% or 50% (depending on which buff). This is not reflected in the `classResources` field of the event, which breaks rune tracking.
+ */
+export default class ExterminateCostNormalizer extends EventsNormalizer {
+  normalize(events: AnyEvent[]): AnyEvent[] {
+    const newEvents = [];
+    let activeBuff: number | undefined = undefined;
+    for (let event of events) {
+      if (this.isExterminateBuff(event) && event.type === 'applybuff') {
+        activeBuff = event.ability.guid;
+      } else if (
+        this.isExterminateBuff(event) &&
+        event.type === 'removebuff' &&
```

Comment:
Wouldn't this nicer? Keep in mind that you cannot just apply this suggestion without importing the EventType enum aswell.
```suggestion
        event.type === EventType.RemoveBuff &&
```